src/common/Session.cpp

```diff
@@ -151,45 +153,59 @@ namespace SDDM {
 
         qDebug() << "Reading from" << m_fileName;
 
-        QFile file(m_fileName);
-        if (!file.open(QIODevice::ReadOnly))
-            return;
+        QSettings Settings(m_fileName, QSettings::IniFormat);
+        QStringList Locales;
+	Locales << QLocale().name() << QLocale().name().remove(QRegExp(QLatin1String("_.*")));
```

Comment:
what is the second locale for?

Reply:
That is because in desktop files it is often the case that some locales are written with a language code only, while others are written with a language code and a territory code, separated by _.